test/unit/test-frame.js

```diff
@@ -32,4 +32,45 @@ describe('Frame', () => {
       });
     });
   });
+
+  describe('path deduplication', () => {
+    it('fixes duplicated source paths', () => {
+      const samples = [
+        {
+          duped:
+            'https://raw.githubusercontent.com/ampproject/amphtml/2004142326360/extensions/amp-viewer-integration/0.1/messaging/extensions/amp-viewer-integration/0.1/messaging/messaging.js',
+          fixed:
+            'https://raw.githubusercontent.com/ampproject/amphtml/2004142326360/extensions/amp-viewer-integration/0.1/messaging/messaging.js',
+        },
+        {
+          duped:
+            'https://raw.githubusercontent.com/ampproject/amphtml/2004071640410/src/service/src/service/storage-impl.js',
+          fixed:
+            'https://raw.githubusercontent.com/ampproject/amphtml/2004071640410/src/service/storage-impl.js',
+        },
+        {
+          duped:
+            'https://raw.githubusercontent.com/ampproject/amphtml/2004071640410/src/src/chunk.js',
+          fixed:
+            'https://raw.githubusercontent.com/ampproject/amphtml/2004071640410/src/chunk.js',
+        },
+        {
+          duped:
+            'https://raw.githubusercontent.com/ampproject/amphtml/2004071640410/src/src-subdir/chunk.js',
+          fixed:
+            'https://raw.githubusercontent.com/ampproject/amphtml/2004071640410/src/src-subdir/chunk.js',
+        },
+        {
+          duped: 'unexpected-token.js',
+          fixed: 'unexpected-token.js',
+        },
+      ];
+
+      samples.forEach(({ duped, fixed }) => {
+        const frame = new Frame('name', duped, '1', '2');
+        console.log(duped, fixed);
```

Comment:
Nit: if you're trying to log individual test cases, you could just procedurally generate the tests:

```js
samples.forEach({ duped, fixed }, index) => {
  it(`fixes duplicated source paths: ${index}`, () => {
    // do test here.
  });
});
```

Reply:
In Jest there's an `it.each` method for this purpose, but I didn't see a mocha equivalent 😕. I'll make this change and submit.